.github/workflows/build.yaml

```diff
@@ -66,7 +66,30 @@ jobs:
       node_type: "gpu-v100-latest-1"
       run_script: "ci/build_docs.sh"
       sha: ${{ inputs.sha }}
+  wheel-build-libraft:
+    secrets: inherit
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-build.yaml@branch-25.02
+    with:
+      build_type: ${{ inputs.build_type || 'branch' }}
+      branch: ${{ inputs.branch }}
+      sha: ${{ inputs.sha }}
+      date: ${{ inputs.date }}
+      script: ci/build_wheel_libraft.sh
+      # build for every combination of arch and CUDA version, but only for the latest Python
+      matrix_filter: group_by([.ARCH, (.CUDA_VER|split(".")|map(tonumber)|.[0])]) | map(max_by(.PY_VER|split(".")|map(tonumber)))
+  wheel-publish-libraft:
+    needs: wheel-build-libraft
+    secrets: inherit
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-publish.yaml@branch-25.02
+    with:
+      build_type: ${{ inputs.build_type || 'branch' }}
+      branch: ${{ inputs.branch }}
+      sha: ${{ inputs.sha }}
+      date: ${{ inputs.date }}
+      package-name: libraft
+      package-type: cpp
   wheel-build-pylibraft:
+    needs: wheel-publish-libraft
```

Comment:
The job downloads from s3, so really this can happen after the libraft build is complete and doesn't have to wait for publish, right?

Reply:
Yes, good point!!!

I'll change that here, and put up PRs across RAPIDS with changes like that.

Across RAPIDS, we have `wheel-build-*` depending on `wheel-publish-*` of its dependencies.

https://github.com/rapidsai/cudf/blob/e272f1e60978801809945c59240dfb02f0aa3b07/.github/workflows/build.yaml#L92-L93

https://github.com/rapidsai/cugraph/blob/6b5db94d9b6ebee41577222d230a36675d3e3ff4/.github/workflows/build.yaml#L90-L91

That extra parallelism should reduce the end-to-end time for publishing nightlies 😁

Reply:
Nice thanks! I'll let you handle doing this elsewhere as time permits.

Reply:
No prob, no prob. I'm gonna wait until we've finished https://github.com/rapidsai/cugraph/pull/4804, in case that uncovers other this-should-change-across-all-of-RAPIDS things, then do a sweep of 1 PR per repo with these changes where necessary.

I'm keeping a list, so far I have:

* nightly / branch builds depending on `*-build` not `*-publish`
* PR builds depending on `*-build` not `*-tests`
* removing `-DDETECT_CONDA_ENV=OFF` from wheel-building scripts
* enforcing `rapids-cmake`'s preferred CMake formatting (https://github.com/rapidsai/raft/pull/2531#discussion_r1917039870)